### PR TITLE
chore: Java generated code pagination test.

### PR DESCRIPTION
**Pull request by kridai**
Closes: DII-268
# Fixes #

Added the Pagination Unit tests for List api calls in Java generated code.
The added unit test would test the functionality of the List api, the pagination handled in the java and error use-case.

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] Run `make test-docker`
- [ ] Verify affected language:
    - [ ] Generate [twilio-go](https://github.com/twilio/twilio-go) from our [OpenAPI specification](https://github.com/twilio/twilio-oai) using the [build_twilio_go.py](./examples/build_twilio_go.py) using `python examples/build_twilio_go.py path/to/twilio-oai/spec/yaml path/to/twilio-go` and inspect the diff
    - [ ] Run `make test` in `twilio-go`
    - [ ] Create a pull request in `twilio-go`
    - [ ] Provide a link below to the pull request
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-oai-generator/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please create a GitHub Issue in this repository.

### CODE REVIEW

File: examples/java/unit-test/rest/TwilioRestTest.java [hunk 1]
```diff
@@ -146,63 +146,125 @@ public void testShouldReadListOfMessages() {
...
+    Map<String, Object> jsonMap = new HashMap<String, Object>();
+    jsonMap.put("code", 20001);
+    jsonMap.put("message", "Invalid PageSize.");
+    jsonMap.put("more_info", "https://www.twilio.com/docs/errors/20001");
+    jsonMap.put("status", 400);
+    JSONObject response = new JSONObject(jsonMap);
+    mockRequest.addQueryParam("To", "9999999999");
+    mockRequest.addQueryParam("From", "4444444444");
+    mockRequest.addQueryParam("PageSize", "0");
+  //  when(twilioRestClient.request(mockRequest)).thenReturn(new Response("{\"messages\":[{\"from\":\"4444444444\", \"to\":\"9999999999\", \"PageSize\":\"5\"}], \"meta\": {\"url\":\"https://api.twilio.com/2010-04-01/Accounts/AC222222222222222222222222222222/Messages.json\", \"next_page_url\":\"\", \"previous_page_url\":\"\", \"first_page_url\":\"\", \"page_size\":5}}", 200));
```
Comment (childish-sambino) [inline examples/java/unit-test/rest/TwilioRestTest.java:260]: Should remove this.

File: examples/java/unit-test/rest/TwilioRestTest.java [hunk 2]
```diff
@@ -146,63 +146,125 @@ public void testShouldReadListOfMessages() {
...
-
-        messageReader.read(twilioRestClient);
-
-        assertNotNull(messageReader);
+            HttpMethod.GET,
+            "api",
+            "/2010-04-01/Accounts/AC222222222222222222222222222222/Messages.json"
+    );
+    ObjectMapper objectMapper = new ObjectMapper();
+    objectMapper.registerModule(new JavaTimeModule());
```
Comment (childish-sambino) [inline examples/java/unit-test/rest/TwilioRestTest.java:249]: Why is this needed? Don't see any time mapping below.

File: examples/java/unit-test/rest/TwilioRestTest.java [hunk 3]
```diff
@@ -146,63 +146,125 @@ public void testShouldReadListOfMessages() {
...
+    mockRequest.addQueryParam("PageSize", "0");
+  //  when(twilioRestClient.request(mockRequest)).thenReturn(new Response("{\"messages\":[{\"from\":\"4444444444\", \"to\":\"9999999999\", \"PageSize\":\"5\"}], \"meta\": {\"url\":\"https://api.twilio.com/2010-04-01/Accounts/AC222222222222222222222222222222/Messages.json\", \"next_page_url\":\"\", \"previous_page_url\":\"\", \"first_page_url\":\"\", \"page_size\":5}}", 200));
+    when(twilioRestClient.request(mockRequest)).thenReturn(new Response(response.toString(), 400));
+    when(twilioRestClient.getObjectMapper()).thenReturn(objectMapper);
+    MessageReader messageReader = new MessageReader("AC222222222222222222222222222222");
+    messageReader.setFrom("4444444444");
+    messageReader.setTo("9999999999");
+    messageReader.setPageSize(0);
+    assertThrows("Invalid PageSize.", ApiException.class, () -> messageReader.read(twilioRestClient));
     }
```
Comment (childish-sambino) [inline examples/java/unit-test/rest/TwilioRestTest.java:268]: Alignment is off.

File: examples/java/unit-test/rest/TwilioRestTest.java [hunk 4]
```diff
@@ -228,6 +290,4 @@ public void testObjectArrayTypeParam() {
 
         assertNotNull(awsCreator);
     }
-}
-
-
+}
```
Comment (childish-sambino) [inline examples/java/unit-test/rest/TwilioRestTest.java:293]: nit: Prefer to always end files with a newline. Most IDEs support this option.

**Review by childish-sambino — APPROVED** [review]

**Review by shrutiburman — APPROVED** [review]
